From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 16 Sep 2020 12:49:39 +0200
Subject: clutter-backend-x11: Don't set the font-dpi computed on X11
 parameters

We already set proprely the font-dpi based on user settings in
MetaSettings at each user change and as part of backend initialization,
so there's no point to set it also during x11 backend post-parsing and
using X11 values, as this may happen at later point and lead to a wrong
clutter font DPI value.

Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1444
---
 clutter/clutter/x11/clutter-backend-x11.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/clutter/clutter/x11/clutter-backend-x11.c b/clutter/clutter/x11/clutter-backend-x11.c
index 697334b..c8cb880 100644
--- a/clutter/clutter/x11/clutter-backend-x11.c
+++ b/clutter/clutter/x11/clutter-backend-x11.c
@@ -262,9 +262,7 @@ clutter_backend_x11_post_parse (ClutterBackend  *backend,
                                 GError         **error)
 {
   ClutterBackendX11 *backend_x11 = CLUTTER_BACKEND_X11 (backend);
-  ClutterSettings *settings;
   Atom atoms[N_ATOM_NAMES];
-  double dpi;
 
   if (_foreign_dpy)
     backend_x11->xdpy = _foreign_dpy;
@@ -304,8 +302,6 @@ clutter_backend_x11_post_parse (ClutterBackend  *backend,
 
   CLUTTER_NOTE (BACKEND, "Getting the X screen");
 
-  settings = clutter_settings_get_default ();
-
   /* add event filter for Cogl events */
   clutter_x11_add_filter (cogl_xlib_filter, backend);
 
@@ -324,11 +320,6 @@ clutter_backend_x11_post_parse (ClutterBackend  *backend,
 
   backend_x11->display_name = g_strdup (clutter_display_name);
 
-  dpi = (((double) DisplayHeight (backend_x11->xdpy, backend_x11->xscreen_num) * 25.4)
-      / (double) DisplayHeightMM (backend_x11->xdpy, backend_x11->xscreen_num));
-
-  g_object_set (settings, "font-dpi", (int) dpi * 1024, NULL);
-
   /* create XSETTINGS client */
   backend_x11->xsettings =
     _clutter_xsettings_client_new (backend_x11->xdpy,
